### ISSUE TITLE
PL/SQL script interpreter wrong end match

### ISSUE THREAD

**Issue by dariopnc** [serge-rider/dbeaver#1836]
When you execute the following code on an Oracle DB using DBeaver 4.1.0 as a script (Alt+X) you get an error

```
DECLARE

	tmp_v varchar2(10):='world';
	
	CURSOR test_c IS
		SELECT *
		FROM dual;
	
BEGIN
	
	FOR cursor_rec IN test_c
	loop
		dbms_output.put_line('Hello '||tmp_v||'!');
	END LOOP;
	
END;
```

Error message (italian localization):
```
SQL Error [6550] [65000]: ORA-06550: riga 14, colonna 10:
PLS-00103: Encountered the symbol "end-of-file" when expecting one of the following:

   ( begin case declare end exception exit for goto if loop mod
   null pragma raise return select update while with
   <un identificativo>
   <un identificativo delimitato da virgolette>
   <una bind variable> << continue close current delete fetch
   lock insert open rollback savepoint set sql execute commit
   forall merge pipe purge

SQL Error [6550] [65000]: ORA-06550: riga 14, colonna 10:
PLS-00103: Encountered the symbol "end-of-file" when expecting one of the following:

   ( begin case declare end exception exit for goto if loop mod
   null pragma raise return select update while with
   <un identificativo>
   <un identificativo delimitato da virgolette>
   <una bind variable> << continue close current delete fetch
   lock insert open rollback savepoint set sql execute commit
   forall merge pipe purge

  ORA-06550: riga 14, colonna 10:
PLS-00103: Encountered the symbol "end-of-file" when expecting one of the following:

   ( begin case declare end exception exit for goto if loop mod
   null pragma raise return select update while with
   <un identificativo>
   <un identificativo delimitato da virgolette>
   <una bind variable> << continue close current delete fetch
   lock insert open rollback savepoint set sql execute commit
   forall merge pipe purge

  ORA-06550: riga 14, colonna 10:
PLS-00103: Encountered the symbol "end-of-file" when expecting one of the following:

   ( begin case declare end exception exit for goto if loop mod
   null pragma raise return select update while with
   <un identificativo>
   <un identificativo delimitato da virgolette>
   <una bind variable> << continue close current delete fetch
   lock insert open rollback savepoint set sql execute commit
   forall merge pipe purge

```

It looks as if the script interpreter doesn't recognize the word `LOOP` after `END` and tries to close the code block there.

Running it highligthing everything and using Ctrl+Enter, works fine. The same code under SQLDeveloper works fine, too.

**Comment by dariopnc** [comment]
I suppose PL/SQL script processing was fixed in [4.0.5](http://dbeaver.jkiss.org/2017/04/16/dbeaver-4-0-5/)

**Comment by serge-rider** [comment]
This is a bug in SQL parser (LOOP/END LOOP aren't recognized as block).
Thanks for report.
